Improve data types page (#745)
data-apis/array-api: spec/draft/API_specification/data_types.rst

.. autosummary::
   :toctree: generated
   :template: method.rst

   __eq__

.. note::
   A conforming implementation of the array API standard may provide and
   support additional data types beyond those described in this specification.
   It may also support additional methods and attributes on dtype objects.

.. note::
   IEEE 754-2019 requires support for subnormal (a.k.a., denormal) numbers, which are useful for supporting gradual underflow. However, hardware support for subnormal numbers is not universal, and many platforms (e.g., accelerators) and compilers support toggling denormals-are-zero (DAZ) and/or flush-to-zero (FTZ) behavior to increase performance and to guard against timing attacks.

   Accordingly, subnormal behavior is left unspecified and, thus, implementation-defined. Conforming implementations may vary in their support for subnormal numbers.

Use of data type objects
------------------------

Data type objects are used as ``dtype`` specifiers in functions and methods
(e.g., ``zeros((2, 3), dtype=float32)``), accessible as ``.dtype`` attribute on
arrays, and used in various casting and introspection functions (e.g.,
``isdtype(x.dtype, 'integral')``).

> ``dtype`` keywords in functions specify the dtype of arrays returned from

Reviewer: ```suggestion
``dtype`` keywords in functions specify the data type of arrays returned from
```

functions or methods. ``dtype`` keywords are not required to affect the data
type used for intermediate calculations or results (e.g., implementors are free

> to use a higher-precision dtype when accumulating values for reductions, as
> long as the returned array has the specified dtype).

Reviewer: ```suggestion
to use a higher-precision data type when accumulating values for reductions, as
long as the returned array has the specified data type).
```

.. note::
   Implementations may provide other ways to specify data types (e.g., ``zeros((2, 3), dtype='f4')``) which are not described in this specification; however, in order to ensure portability, array library consumers are recommended to use data type objects as provided by specification conforming array libraries.

> When arrays with different dtypes or different dtype objects are used together,
> what happens is described in :ref:`type-promotion`.

Reviewer: ```suggestion
See :ref:`type-promotion` for specification guidance describing the rules governing the interaction of two or more data types or data type objects.
```
